src/indexer/contract_state_indexer.rs

```diff
@@ -150,49 +150,103 @@ where
+                store.unsettled_blobs.remove(&tx_id)
+            } else {
+                store.unsettled_blobs.get(&tx_id).cloned()
+            } {
+                Some(tx) => tx,
+                None => {
+                    debug!(cn = %self.contract_name, "🔨 No supported blobs found in transaction: {}", tx);
+                    continue;
+                }
+            };
+
+            let state = store
+                .state
+                .as_mut()
+                .ok_or(anyhow!("No state found for {}", self.contract_name))?;
+
+            for (index, Blob { contract_name, .. }) in tx.blobs.iter().enumerate() {
+                if self.contract_name != *contract_name {
+                    continue;
+                }
+
+                handler(state, &tx, BlobIndex(index), tx_context.clone())?;
+            }
+        }
+        Ok(())
+    }
+
     async fn handle_processed_block(&mut self, block: Block) -> Result<()> {
-        for (_, contract) in block.registered_contracts {
+        for (_, contract) in &block.registered_contracts {
             if self.contract_name == contract.contract_name {
-                self.handle_register_contract(contract).await?;
+                self.handle_register_contract(contract.clone()).await?;
             }
         }
 
         if !block.txs.is_empty() {
             debug!(handler = %self.contract_name, "🔨 Processing block: {}", block.block_height);
         }
 
-        for (tx_id, tx) in block.txs {
-            if let TransactionData::Blob(tx) = tx.transaction_data {
-                self.handle_blob(tx_id, tx).await?;
+        for (tx_id, tx) in &block.txs {
+            if let TransactionData::Blob(tx) = &tx.transaction_data {
+                self.handle_blob(tx_id.clone(), tx.clone()).await?;
             }
         }
 
-        for s_tx in block.successful_txs {
-            let dp_hash = block
-                .dp_parent_hashes
-                .get(&s_tx)
-                .context(format!(
-                    "No parent data proposal hash present for successful tx {}",
-                    s_tx.0
-                ))?
-                .clone();
-            let lane_id = block
-                .lane_ids
-                .get(&s_tx)
-                .context(format!("No lane id found for TX hash {}", dp_hash.0))?
-                .clone();
-            self.settle_tx(
-                &TxId(dp_hash, s_tx),
-                TxContext {
-                    lane_id,
-                    block_hash: block.hash.clone(),
-                    block_height: block.block_height,
-                    timestamp: block.block_timestamp.clone(),
-                    chain_id: HYLE_TESTNET_CHAIN_ID, // TODO: make it configurable
-                },
-            )
-            .await?;
-        }
+        self.handle_tx_list(
+            &block
+                .txs
+                .iter()
+                .map(|(tx, _)| tx.1.clone())
+                .collect::<Vec<_>>(),
```

Comment:
Ici cf autre comm, ça doit être possible de passer l'itérateur direct en argument